specs/sql/README.md

```diff
@@ -335,18 +335,133 @@ contain only column names. An error is raised if more than one
-> evaluating whether to allow `ALTER TABLE` only in the case of *adding
-> new columns*.
+#### Integer Primary Key
+
+All rows within Tableland tables have a 64-bit signed integer key that
+uniquely identifies the row within its table. This integer is usually
+called the `ROWID`. The `ROWID` value can be accessed using one of the
+special case-independent names `"rowid"`, `"oid"`, or `"_rowid_"` in
+place of a column name. If a table contains a user defined column named
+`"rowid"`, `"oid"` or `"_rowid_"`, then that name always refers the
+explicitly declared column and cannot be used to retrieve the integer
+`ROWID` value.
+
+The data for Tableland tables are stored in sorted order, by `ROWID`.
+This means that retrieving or sorting records by `ROWID` is fast.
+Searching for a record with a specific `ROWID`, or for all records with
+`ROWID`s within a specified range is around twice as fast as a similar
+search made by specifying any other `PRIMARY KEY`. This `ROWID` sorting
+is also required for sub-queries and other SQL features on Tableland, to
+ensure deterministic ordering of results.
+
+With one exception noted below, if a table has a primary key that
+consists of a single column and the declared type of that column is
+`INTEGER`, then the column becomes an alias for the `ROWID`. Such a
+column is usually referred to as an "integer primary key". A
+`PRIMARY KEY` column only becomes an integer primary key if the declared
+type name is exactly `INTEGER`. Other integer type names like `INT`
+causes the primary key column to behave as an ordinary table column with
+integer affinity and a unique index, not as an alias for the `ROWID`.
+
+In the above case of an integer primary key, there is an additional
+implied `AUTOINCREMENT` constraint, which forces the integer primary key
+to behave as if it were specified with
+`INTEGER PRIMARY KEY AUTOINCREMENT`. See [Autoincrement](#autoincrement)
+for further details.
+
+The exception mentioned above is that if the declaration of a column
+with declared type `INTEGER` includes an `PRIMARY KEY DESC` clause, it
+does not become an alias for the `ROWID` and is not classified as an
+integer primary key. In pratice, this means that the following three
+table declarations all cause the column "x" to be an alias for the
+`ROWID` (an integer primary key):
+
+-   `CREATE TABLE t(x INTEGER PRIMARY KEY ASC, y, z);`
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x ASC));`
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x DESC));`
+
+But the following declaration does not result in "x" being an alias for
+the `ROWID`:
+
+-   `CREATE TABLE t(x INTEGER PRIMARY KEY DESC, y, z);`
+
+Given this, and the implied autoincrement behavior, if the primary key
+is defined at the end as a table constraint and not as a column
+constraint:
+
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x DESC));`
+
+this will actually be transformed (behind the scenes) by the Tableland
```

Comment:
As discussed in Discord, we are not going to force `AUTOINCREMENT` in this case.